src/Microsoft.PowerShell.ConsoleHost/host/msh/StartTranscriptCmdlet.cs

```diff
@@ -228,7 +227,19 @@ protected override void BeginProcessing()
                         System.IO.File.Delete(effectiveFilePath);
                     }
                 }
-
+                else 
+                {
+                    // check if the path resolves to an existing or non-existing directory. 
+                    string baseDirectory = System.IO.Path.GetDirectoryName(effectiveFilePath);
+                    if(Directory.Exists(effectiveFilePath) || baseDirectory.Equals(effectiveFilePath.TrimEnd(System.IO.Path.DirectorySeparatorChar)))
```

Comment:
There is a Path variable in this class. Path here is wrongly interpreted as a string which results in compilation error.
Hence the usage of System.IO.Path.

Good catch for the baseDirectory null check. Will correct this.